lib/ansible/modules/network/avi/avi_controllerproperties.py

```diff
@@ -0,0 +1,331 @@
+#!/usr/bin/python
+#
+# Created on Aug 25, 2016
+# @author: Gaurav Rastogi (grastogi@avinetworks.com)
+#          Eric Anderson (eanderson@avinetworks.com)
+# module_check: not supported
+# Avi Version: 16.3.4
+#
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+HAS_AVI = True
+try:
+    from avi.sdk.utils.ansible_utils import (
+        avi_ansible_api, avi_common_argument_spec)
+except ImportError:
+    HAS_AVI = False
+
+# Comment: import * is to make the modules work in ansible 2.0 environments
+# from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import *
+
+DOCUMENTATION = '''
+---
+module: avi_controllerproperties
+author: Gaurav Rastogi (grastogi@avinetworks.com)
+
+short_description: ControllerProperties Configuration
+description:
+    - This module is used to configure ControllerProperties object
+    - more examples at U(https://github.com/avinetworks/avi-ansible-samples)
+requirements: [ avisdk ]
+version_added: 2.3
+options:
+    state:
+        description:
+            - The state that should be applied on the entity.
+        required: false
+        default: present
+        choices: ["absent","present"]
+    allow_unauthenticated_apis:
+        description:
+            - Allow unauthenticated access for special apis. 
+        default: False
+        type: bool
```

Comment:
We don't define the `type:` in the `DOCUMENTATION` block, only in `RETURNS`, so these can be removed from all modules.